core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -621,6 +615,80 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   }
 }
 
-private[spark] object CoarseGrainedSchedulerBackend {
+private[spark] object CoarseGrainedSchedulerBackend extends Logging {
   val ENDPOINT_NAME = "CoarseGrainedScheduler"
+
+  // abort TaskSetManager without exception
+  private def abortTaskSetManager(
+      scheduler: TaskSchedulerImpl,
+      taskId: Long,
+      msg: => String,
+      exception: Option[Throwable] = None): Unit = scheduler.synchronized {
+    scheduler.taskIdToTaskSetManager.get(taskId).foreach { taskSetMgr =>
+      try {
+        taskSetMgr.abort(msg, exception)
+      } catch {
+        case e: Exception => logError("Exception in error callback", e)
+      }
+    }
+  }
+
+  private def isZombieTaskSetManager(
+    scheduler: TaskSchedulerImpl,
+    taskId: Long): Unit = scheduler.synchronized {
+    !scheduler.taskIdToTaskSetManager.get(taskId).exists(_.isZombie)
+  }
+
+  private def getTaskSetManager(
+    scheduler: TaskSchedulerImpl,
+    taskId: Long): Option[TaskSetManager] = scheduler.synchronized {
+    scheduler.taskIdToTaskSetManager.get(taskId)
+  }
+
+  private[scheduler] def prepareSerializedTask(
+    scheduler: TaskSchedulerImpl,
+    task: TaskDescription,
+    abortSet: HashSet[TaskSetManager],
+    maxRpcMessageSize: Long): ByteBuffer = {
+    var serializedTask: ByteBuffer = null
+    if (abortSet.isEmpty || !getTaskSetManager(scheduler, task.taskId).exists(_.isZombie)) {
```

Comment:
This is an optimization code, in most cases abortSet is empty and the `!getTaskSetManager(scheduler, task.taskId).exists(_.isZombie)` is not called.
 I changed it into a more readable code.

see https://github.com/apache/spark/pull/15505/commits/2f4b3f955cad0fe7546de54737c19500b90ad67d